src/sinks/aws_cloudwatch_logs/mod.rs

```diff
@@ -162,28 +183,36 @@ impl Service<PartitionInnerBuffer<Vec<Event>, CloudwatchKey>> for CloudwatchLogs
 
         let RequestConfig {
             timeout_secs,
-            in_flight_limit,
             rate_limit_duration_secs,
             rate_limit_num,
+            retry_attempts,
+            retry_backoff_secs,
         } = self.request_config;
 
         let svc = if let Some(svc) = &mut self.clients.get_mut(&key) {
             svc.clone()
         } else {
             let svc = {
+                let policy = FixedRetryPolicy::new(
+                    retry_attempts,
+                    Duration::from_secs(retry_backoff_secs),
+                    CloudwatchRetryLogic,
+                );
+
                 let cloudwatch = CloudwatchLogsSvc::new(&self.config, &key).unwrap();
                 let timeout = Timeout::new(cloudwatch, Duration::from_secs(timeout_secs));
 
-                // TODO: add Buffer/Retry here
+                let buffer = Buffer::new(timeout, 1);
+                let retry = Retry::new(policy, buffer);
 
                 let rate = RateLimit::new(
-                    timeout,
+                    retry,
                     Rate::new(
                         rate_limit_num,
                         Duration::from_secs(rate_limit_duration_secs),
                     ),
                 );
-                let concurrency = ConcurrencyLimit::new(rate, in_flight_limit);
+                let concurrency = ConcurrencyLimit::new(rate, 1);
```

Comment:
This kinda feels redundant, since we already need to prevent multiple concurrent requests from `poll_ready` for token reasons.